packages/auto-label/src/auto-label.ts

```diff
@@ -249,15 +245,102 @@ handler.autoLabelOnPR = async function autoLabelOnPR(
       logger.info(
         `Language label added to PR #${pull_number} in ${owner}/${repo} is ${language_label}`
       );
+      new_labels.push(language_label);
+    }
+  }
+
+  // Update all labels gathered so far if any
+  if (new_labels.length) {
+    logger.info(
+      `Labels added to PR# ${pull_number} in ${owner}/${repo} are ${JSON.stringify(
+        new_labels
+      )}`
+    );
+    await context.octokit.issues.addLabels({
+      owner,
+      repo,
+      issue_number: pull_number,
+      labels: new_labels,
+    });
+  }
+};
+
+/**
+ * Function updating a stale label on pull request based on configuration values.
+ * Removes previous staleness label if exists
+ */
+async function updateStalenessLabel(
+  context: Context<'pull_request'> | Context<'issues'>,
+  owner: string,
+  repo: string,
+  config: Config
+) {
+  // If user has turned on stale labels by configuring {staleness: {pullrequest: true, old: 60, critical: 120}}
+  // By default, this feature is turned off
+  if (!config.staleness?.pullrequest) {
+    return;
+  }
+  const response = await context.octokit.rest.pulls.list({
+    owner: owner,
+    repo: repo,
+  });
+  if (!response || !response.data) {
+    logger.info(`No active PRs available in ${owner}/${repo}...`);
+    return;
+  }
+
+  let old = config.staleness?.old || helper.DEFAULT_DAYS_TO_STALE;
+  let critical = config.staleness?.critical || helper.DEFAULT_DAYS_TO_STALE * 2;
+  let label = null;
+
+  // Make sure that config is right and if not, set defaults
+  if (old > critical || critical > helper.MAX_DAYS || old < 0) {
+    logger.info(
+      `The staleness config is wrong, old = ${old}, critical = ${critical}, fixing...`
+    );
+    old = helper.DEFAULT_DAYS_TO_STALE;
+    critical = helper.DEFAULT_DAYS_TO_STALE * 2;
+  }
+
+  for (const pull of response.data) {
+    logger.info(
+      `Checking staleness in PR #${pull.number} in ${owner}/${repo}...`
+    );
+    const staleLabel = helper.fetchLabelByPrefix(
+      pull.labels,
+      helper.STALE_PREFIX
+    );
+
+    if (helper.isExpiredByDays(pull.created_at, critical)) {
+      label = `${helper.STALE_PREFIX} ${helper.CRITICAL_LABEL}`;
+    } else if (helper.isExpiredByDays(pull.created_at, old)) {
+      label = `${helper.STALE_PREFIX} ${helper.OLD_LABEL}`;
+    }
+
+    if (label && label !== staleLabel?.name) {
+      // We are going to update a label now, remove an old one if exists
+      if (staleLabel) {
+        logger.info(
+          `Deleting ${staleLabel.name!} in ${owner}/${repo}/${pull.number}...`
+        );
+        context.octokit.issues.removeLabel({
+          owner,
+          repo,
+          issue_number: pull.number,
+          name: staleLabel.name!,
+        });
+      }
+
+      logger.info(`Adding ${label} to ${owner}/${repo}/${pull.number}...`);
```

Comment:
It would be interesting to emit a metric here, so that we can make a dashboard of stale PRs over time, here's an example from merge-on-green:

```js
    logger.metric('merge_on_green.mergeability_sample', {
      repo: `${owner}/${repo}/`,
      number: pr,
      mergeable: prInfo.mergeable,
      mergeable_state: prInfo.mergeable_state,
      checkReviews,
      checkStatus,
    });
```

Reply:
yep, makes sense